### PR TITLE
[SDL 0273] WebEngine Projection mode

### LINKED ISSUE #1681: [SDL 0273] WebEngine Projection mode
Proposal: [WebEngine Projection mode](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0273-webengine-projection-mode.md)

> This proposal is created based on discussion in https://github.com/smartdevicelink/sdl_evolution/issues/767.
It extends [SDl-0240 - WebEngine support for SDL JavaScript](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0240-sdl-js-pwa.md) to allow using the WebEngine app's web view for the HMI in addition to the other available SDL templates.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/912

Steering Committee Decision:

>The Steering Committee voted to accept this proposal with the revisions outlined in [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/912#issuecomment-637749152).

> The proposal .md file was updated to reflect these revisions on 6/8/2020.

### PR DESCRIPTION
**Pull request by yoooriii**
Fixes #1681 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR
- [x] I have run the unit tests with this PR

#### Unit Tests
- Corresponded tests to test updated enums were added

### Summary

- Added an enum item to SDLAppHMIType
- Added an enum item to SDLAppInterfaceUnregisteredReason
- Added an enum item to SDLPredefinedLayout
- Also corresponded tests were updated

### Changelog
[detailed description on sdl evolution](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0273-webengine-projection-mode.md)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by yoooriii** [comment]
@kshala-ford  @mrapitis @abyzhynar 
Could you please review the PR?

File: SmartDeviceLink/SDLAppInterfaceUnregisteredReason.h [hunk 1]
```diff
@@ -73,3 +73,9 @@ extern SDLAppInterfaceUnregisteredReason const SDLAppInterfaceUnregisteredReason
 ///
 /// @since RPC 4.1
 extern SDLAppInterfaceUnregisteredReason const SDLAppInterfaceUnregisteredReasonUnsupportedHMIResource;
+
+
+/// By getting this value, SDL should unregister the application to allow the HMI to close the application.
+///
+/// @since RPC 6.0
```
Comment (kshala-ford) [inline SmartDeviceLink/SDLAppInterfaceUnregisteredReason.h:80]: I think this description needs to be updated to match the requested description change in rpc_spec issue https://github.com/smartdevicelink/rpc_spec/pull/270. Also version number should be 7.0

File: SmartDeviceLink/SDLPredefinedLayout.h [hunk 2]
```diff
@@ -108,3 +108,10 @@ extern SDLPredefinedLayout const SDLPredefinedLayoutDoubleGraphicWithSoftButtons
  A layout with only a single large graphic
  */
 extern SDLPredefinedLayout const SDLPredefinedLayoutLargeGraphicOnly;
+
+
+/**
+ Custom root template allowing in-vehicle WebEngine applications with
+ appropriate permissions to show the application's own web view.
+ */
```
Comment (kshala-ford) [inline SmartDeviceLink/SDLPredefinedLayout.h:116]: Since SmartDeviceLink 7.0

File: SmartDeviceLink/SDLAppHMIType.h [hunk 3]
```diff
@@ -69,3 +69,9 @@ extern SDLAppHMIType const SDLAppHMITypeSystem;
 /**
  * Remote control */
 extern SDLAppHMIType const SDLAppHMITypeRemoteControl;
+
+
+/**
+ * WebEngine Projection mode
+ */
```
Comment (kshala-ford) [inline SmartDeviceLink/SDLAppHMIType.h:76]: Since 7.0

**Comment by yoooriii** [comment]
Hello, I have fixed the feedback issues, @kshala-ford could you please review if it is ok? Thank you.

**Comment by SilvioDC-Ford** [comment]
@theresalech this PR is ready for review and is approved by Ford.

**Comment by theresalech** [comment]
@yoooriii can you please update the PR description to note that the PR is ready for review?  It currently says `This PR is **not ready** for review.`  Additionally, please update PR so that it is no longer a `Draft`.  Thank you!

**Comment by yoooriii** [comment]
> @yoooriii can you please update the PR description to note that the PR is ready for review? It currently says `This PR is **not ready** for review.` Additionally, please update PR so that it is no longer a `Draft`. Thank you!

Yes, it is done. Thank you.

**Comment by joeljfischer** [comment]
This will be merged after the v6.7.0 release

**Pull request merged**